hnn_core/network.py

```diff
@@ -763,136 +763,49 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
-                drive_conn_by_cell[cellname] = drive_conn
-        else:
-            drive_conn = dict()
-            drive_conn['location'] = location
-
-            # NB list! This is used later in _parnet_connect
-            drive_conn['src_gids'] = range(src_gid_curr,
-                                           src_gid_curr + n_drive_cells)
-            src_gid_curr += n_drive_cells
-            drive_conn['target_gids'] = list()  # fill in below
-            for cellname in target_populations:
-                drive_conn['target_gids'] = self.gid_ranges[cellname]
-                drive_conn['target_type'] = cellname
-                drive_conn_by_cell[cellname] = drive_conn.copy()
-
-        for cellname in target_populations:
-            for receptor, weights in weights_by_receptor.items():
-                drive_conn_by_cell[cellname][receptor] = dict()
-                if cellname in weights:
-                    drive_conn_by_cell[cellname][receptor][
-                        'lamtha'] = space_constant
-                    if isinstance(synaptic_delays, float):
-                        drive_conn_by_cell[cellname][receptor][
-                            'A_delay'] = synaptic_delays
-                    elif isinstance(synaptic_delays, dict):
-                        drive_conn_by_cell[cellname][receptor][
-                            'A_delay'] = synaptic_delays[cellname]
-                    drive_conn_by_cell[cellname][receptor][
-                        'A_weight'] = weights[cellname]
-
-        return drive_conn_by_cell, range(self._n_gids, src_gid_curr)
+        for receptor in receptors:
+            if cell_specific:
+                # Set the starting index for source gids
+                # This will be updated depending on the number of target cells
+                # for each cell type
+                src_idx = 0
+                for target_cell_type in weights_by_receptor[receptor]:
+                    target_gids = list(self.gid_ranges[target_cell_type])
+                    target_gids_nested = [[target_gid] for
+                                          target_gid in target_gids]
+                    src_idx_end = src_idx + len(target_gids)
+                    src_gids_by_type = (list(self.gid_ranges[name])
+                                        [src_idx:src_idx_end])
+                    weights = weights_by_receptor[receptor][target_cell_type]
+                    delays = synaptic_delays[target_cell_type]
+                    self.add_connection(src_gids_by_type, target_gids_nested,
+                                        location, receptor, weights, delays,
+                                        space_constant)
+                    src_idx = src_idx_end
+            else:
+                for target_cell_type in weights_by_receptor[receptor]:
+                    target_gids = list(self.gid_ranges[target_cell_type])
+                    src_gids = list(self.gid_ranges[name])
```

Comment:
you should be able to give the name `target_cell_type` to `self.add_connection` and it should do the all-to-all connection for you

Reply:
Yes, that's the next step but we'll also need to set the `weight` and `delay` parameters with null/zero values since some of these are currently undefined. We'll then need to adjust the tests so that they don't fail when the weight and delay parameters defined in `net.connectivity` don't match those defined by the user.